compiler/modules/front.syntax/astString.bal

```diff
@@ -282,9 +282,20 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
         if wrap != false {
             w.push("(");
         }
-        typeDescToWords(w, td.rest, true);
+        TypeDesc? rest = td.rest;
+        if rest != () {
+            typeDescToWords(w, rest, true);
+        }
         w.push(CLING);
-        w.push("[", "]");
+        SimpleConstExpr? length = td.length;
+        if length != () {
+            w.push("[", CLING);
+            exprToWords(w, length);
+            w.push(CLING, "]");
+        }
+        else {
+            w.push("[", "]");
```

Comment:
I would factor this out of if-else

Reply:
As in extract a function using this `if-else` block?

Reply:
No, I meant putting `w.push("[");` before if and `w.push("]");` and getting rid of `else`.

Reply:
Oh, I didn't see that at that time, fixed, thanks.